pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -0,0 +1,274 @@
+        items = items.replace("',", ":")
+        items = items.replace(")", "")
+
+        gate_type_str = ", ".join(
+            [f"'{gate_name}': {count}" for gate_name, count in self.gate_types.items()]
+        )
+        items += "\ngate_types:\n{" + gate_type_str + "}"
+        return items
+
+    def _ipython_display_(self):
+        """Displays __str__ in ipython instead of __repr__"""
+        print(str(self))
+
+
+def add_in_series(first: Resources, other: Resources, in_place=False) -> Resources:
+    r"""Add two resources assuming the circuits are executed in series.
+
+    Args:
+        first (Resources): first resource object to combine
+        other (Resources): other resource object to combine with
+        in_place (bool): determines if the first Resources are modified in place (default False)
+
+    Returns:
+        Resources: combined resources
+    """
+    new_wires = max(first.num_wires, other.num_wires)
+    new_gates = first.num_gates + other.num_gates
+    new_gate_types = _combine_dict(first.gate_types, other.gate_types, in_place=in_place)
+
+    if in_place:
+        first.num_wires = new_wires
+        first.num_gates = new_gates
+        return first
+
+    return Resources(new_wires, new_gates, new_gate_types)
+
+
+def add_in_parallel(first: Resources, other: Resources, in_place=False) -> Resources:
+    r"""Add two resources assuming the circuits are executed in parallel.
+
+    Args:
+        first (Resources): first resource object to combine
+        other (Resources): other resource object to combine with
+        in_place (bool): determines if the first Resources are modified in place (default False)
+
+    Returns:
+        Resources: combined resources
+    """
+    new_wires = first.num_wires + other.num_wires
+    new_gates = first.num_gates + other.num_gates
+    new_gate_types = _combine_dict(first.gate_types, other.gate_types, in_place=in_place)
+
+    if in_place:
+        first.num_wires = new_wires
+        first.num_gates = new_gates
+        return first
+
+    return Resources(new_wires, new_gates, new_gate_types)
+
+
+def mul_in_series(first: Resources, scaler: int, in_place=False) -> Resources:
+    r"""Multiply two resources assuming the circuits are executed in series.
+
+    Args:
+        first (Resources): first resource object to combine
+        scaler (int): integer value to scale the resources by
+        in_place (bool): determines if the first Resources are modified in place (default False)
+
+    Returns:
+        Resources: combined resources
+    """
+    new_gates = scaler * first.num_gates
+    new_gate_types = _scale_dict(first.gate_types, scaler, in_place=in_place)
+
+    if in_place:
+        first.num_gates = new_gates
+        return first
+
+    return Resources(first.num_wires, new_gates, new_gate_types)
+
+
+def mul_in_parallel(first: Resources, scaler: int, in_place=False) -> Resources:
+    r"""Multiply two resources assuming the circuits are executed in parallel.
```

Comment:
Maybe reword this since we aren't multiplying two resources, we are multiplying a resource by a scalar.

Reply:
Added